src/wrappers/gcc_wrapper.cpp

```diff
@@ -289,8 +311,8 @@ string_list_t gcc_wrapper_t::get_relevant_arguments() {
       // to binary object files)?
       const auto first_two_chars = arg.substr(0, 2);
       const bool is_unwanted_arg =
-          ((first_two_chars == "-I") || (first_two_chars == "-D") || (first_two_chars == "-M") ||
-           (arg.substr(0, 10) == "--sysroot=") || is_source_file(arg));
+          ((first_two_chars == "-I") || (uses_defines_in_preprocess() && first_two_chars == "-D") ||
```

Comment:
With Direct Mode here (before and after this change), isn't there a bug where command line defines are not included in the hash? `get_relevant_arguments()` might need to be different for direct mode where the preprocessor is not implicitly consuming command line arguments to include in the resulting preprocessed source hash.

Reply:
Actually the complete command line is hashed for direct mode, so `get_relevant_arguments()` is more or less irrelevant for direct mode. See: https://github.com/mbitsnbites/buildcache/blob/master/src/wrappers/program_wrapper.cpp#L144

It's a bit confusing, and not 100% optimal, so I may revisit this later.

Reply:
Yeah, was confused by https://github.com/mbitsnbites/buildcache/blob/master/src/wrappers/program_wrapper.cpp#L126 where it also uses `get_relevant_arguments()` for direct mode. Didn't look further down :)